### PR TITLE
ajout de statut

### PR DESCRIPTION
**Pull request by IkrameHAD**

### CODE REVIEW

File: hceres/src/main/java/org/centrale/hceres/items/TypeActivity.java [hunk 1]
```diff
@@ -90,7 +90,8 @@ public enum IdTypeActivity {
         SEI_COMPANY_CREATION(38),
         SEI_LEAD_CONSORTIUM_INDUSTRY(39),
         SEI_INDUSTRIAL_R_D_CONTRACT(40),
-        SEI_CLINICAL_TRIAL(41);
+        SEI_CLINICAL_TRIAL(41),
+        TEAM(42);
```
Comment (Aziz19979) [inline hceres/src/main/java/org/centrale/hceres/items/TypeActivity.java]: Why add team as new activity ?
->remove

File: hceres/src/main/java/org/centrale/hceres/service/TeamService.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        team.setActivityList((List<Activity>) activity);
+        activity.setTeamList((List<Team>) team);
+        activity.setIdTypeActivity(TypeActivity.IdTypeActivity.TEAM.getId());
+
+        // get list of researcher doing this activity - currently only one is sent
+        activity.setResearcherList(Collections.singletonList(new Researcher(RequestParser.getAsInteger(request.get("researcherId")))));
+
+        activity = activityRepo.save(activity);
+        return activity;
+        */
```
Comment (Aziz19979) [inline hceres/src/main/java/org/centrale/hceres/service/TeamService.java]: Remove useless commented code,
laboratoryId attribute is not set which is mandatory for team creation :

>         team.setLaboratoryId(RequestParser.getAsInteger(request.get("laboratoryId")));

and use teamRepo to save the new created team
>         teamRepo.save(team);

test of creation should be tested in junit test or postman request or react form.

File: doc/code-generator/entities/Team/TeamForm.csv [hunk 3]
```diff
@@ -0,0 +1,5 @@
+data Label, html type, variableName, initialVariableNameValue, jsonPath
+teamName, string, teamName, "", team.teamName
+teamCreation, date, teamCreation, "", team.teamCreation
+teamEnd, date, teamEnd, "", team.teamEnd
+teamLastReport, date, teamLastReport, "", team.teamLastReport
```
Comment (Aziz19979) [inline doc/code-generator/entities/Team/TeamForm.csv]: data label column should be translated into french,
for example:
Nom de l'équipe
Date de création d'équipe
Date de fin d'équipe
dernier rapport de l'équipe

File: hceres-frontend/src/components/Researcher/Researcher.js [hunk 4]
```diff
@@ -159,7 +159,23 @@ class Researcher extends Component {
...
+                    if(row.contract.length > 0) {
+                        statusName = row.contract[0].status.nameStatus
+                        console.log(row.contract[0])
+                    } else {
+                        statusName = 'No contract'
+                    }
+                    return statusName;
+                },
+            },
+                {
```
Comment (Aziz19979) [inline hceres-frontend/src/components/Researcher/Researcher.js]: make parenthesis on same line,
....
                    return statusName;
                },
            }, {
                dataField: 'actionColumn',

File: hceres/src/main/java/org/centrale/hceres/items/BelongsTeam.java [hunk 5]
```diff
@@ -60,4 +60,5 @@ public class BelongsTeam implements Serializable {
     @JoinColumn(name = "team_id", referencedColumnName = "team_id", insertable = false, updatable = false)
     @ManyToOne
     private Team team;
+
```
Comment (Aziz19979) [inline hceres/src/main/java/org/centrale/hceres/items/BelongsTeam.java]: try to impact as less files as you can when commit, remove addition of white line

File: hceres/src/main/java/org/centrale/hceres/service/TeamService.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    private ActivityRepository activityRepo;
+
+    /**
+     * permet de retourner la liste
+     */
+    /**
+    public List<Activity> getTeams() {
+        return activityRepo.findByIdTypeActivity(TypeActivity.IdTypeActivity.TEAM.getId());
+    }
+    */
```
Comment (Aziz19979) [inline hceres/src/main/java/org/centrale/hceres/service/TeamService.java]: remove useless activities related code

File: hceres/src/main/java/org/centrale/hceres/controller/TeamController.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+
+import org.centrale.hceres.items.Activity;
+import org.centrale.hceres.items.Team;
+import org.centrale.hceres.service.TeamService;
+import org.centrale.hceres.util.RequestParseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.text.ParseException;
```
Comment (Aziz19979) [inline hceres/src/main/java/org/centrale/hceres/controller/TeamController.java]: turn on sonar lint to fix code smell.
for example here remove unused import 
import org.centrale.hceres.items.Activity;
import java.text.ParseException;
import java.util.Optional

**Review by Aziz19979 — CHANGES_REQUESTED** [review]

**Review by Ahmad-Said — APPROVED** [review]

**Review by Aziz19979 — APPROVED** [review]

**Pull request merged**